src/trusted/service_runtime/lind_syscalls.c

```diff
@@ -720,95 +768,105 @@ static int NaClCopyZStr(struct NaClApp *nap,
+    if ((inNum && !inArgs) || (outNum && !outArgs)) {
         NaClLog(LOG_ERROR, "NaClSysLindSyscall: in/out arguments are NULL\n");
         retval = -NACL_ABI_EFAULT;
         goto cleanup;
     }
 
-    if(inNum && !NaClCopyInFromUser(nap, inArgSys, (uintptr_t)inArgs, sizeof(LindArg)*inNum)) {
-        NaClLog(LOG_ERROR, "NaClSysLindSyscall: invalid input argument address\n");
+    if (inNum && !NaClCopyInFromUser(nap, inArgSys, (uintptr_t)inArgs, sizeof(LindArg)*inNum)) {
         retval = -NACL_ABI_EFAULT;
+        NaClLog(LOG_ERROR, "NaClSysLindSyscall: invalid input argument address\n");
         goto cleanup;
     }
 
-    for(uint32_t i=0; i<inNum; ++i) {
-        if(inArgSys[i].type != AT_INT) {
-            if(inArgSys[i].ptr) {
-                argSysAddr = NaClUserToSysAddrRange(nap, (uintptr_t)inArgSys[i].ptr, inArgSys[i].len);
+    for (uint32_t j = 0; j<inNum; ++j) {
+        if(inArgSys[j].type != AT_INT) {
+            if(inArgSys[j].ptr) {
+                argSysAddr = NaClUserToSysAddrRange(nap, (uintptr_t)inArgSys[j].ptr, inArgSys[j].len);
                 if(kNaClBadAddress == argSysAddr) {
                     NaClLog(LOG_ERROR, "NaClSysLindSyscall: invalid input data address\n");
                     retval = -NACL_ABI_EFAULT;
                     goto cleanup;
                 }
-                inArgSys[i].ptr = argSysAddr;
-            } else if(inArgSys[i].type == AT_DATA || inArgSys[i].type == AT_STRING) {
+                inArgSys[j].ptr = argSysAddr;
+            } else if(inArgSys[j].type == AT_DATA || inArgSys[j].type == AT_STRING) {
                 NaClLog(LOG_ERROR, "NaClSysLindSyscall: mandatory input is NULL\n");
                 retval = -NACL_ABI_EFAULT;
                 goto cleanup;
             }
         }
     }
 
-    if(outNum && !NaClCopyInFromUser(nap, outArgSys, (uintptr_t)outArgs, sizeof(LindArg)*outNum)) {
+    if (outNum && !NaClCopyInFromUser(nap, outArgSys, (uintptr_t)outArgs, sizeof(LindArg)*outNum)) {
         NaClLog(LOG_ERROR, "NaClSysLindSyscall: invalid output argument address\n");
         retval = -NACL_ABI_EFAULT;
         goto cleanup;
     }
 
-    for(uint32_t i=0; i<outNum; ++i) {
-		//mandatory output address is zero
-		if(outArgSys[i].type == AT_INT || (!outArgSys[i].ptr && outArgSys[i].type == AT_DATA)) {
-			retval = -NACL_ABI_EFAULT;
-			goto cleanup;
-		}
+    for (uint32_t j = 0; j < outNum; ++j) {
+        /* mandatory output address is zero */
+        if(outArgSys[j].type == AT_INT || (!outArgSys[j].ptr && outArgSys[j].type == AT_DATA)) {
+                retval = -NACL_ABI_EFAULT;
+                goto cleanup;
+        }
     }
 
-    /*
-    // For debugging
-    for(int i=0; i<(int)inNum; ++i) {
-        DumpArg(&inArgSys[i]);
+    /* yiwen: handle lind_pipe here.
+     *        the pipe() call should initialize the pipe buffer
+     *        and returns two fds for the pipe
+     */
+    if (callNum == 61) {
+        int data[2] = {9001, 9002};
```

Comment:
So this is my old testing code, which used hardcoded 9001 and 9002 as fds for our own pipe. If we are using different fds for our pipes now, this should be modified. If we are not using this code now, I think we should put this piece of code in a comment and mark it as "unused".

Reply:
This has been removed as per @JustinCappos in my other branches, but it does seem better to just not add it at all, yeah. Going to just rip it out (of course with testing to be sure this doesn't break anything).